src/corredor/helpers/messaging.ts

```diff
@@ -1,15 +1,32 @@
 /* eslint-disable */
 
-import { Channel } from '../../messaging/types/channel'
-import { User } from '../../system/types/user'
-import { extractID, genericPermissionUpdater, isFresh } from './shared'
+import { extractID, genericPermissionUpdater, isFresh, Rule } from './shared'
+import { Messaging as MessagingAPI } from '../../api-clients'
+import { Channel, Message } from '../../messaging/'
+import { User } from '../../system'
+
+type StrIndex<TValue> = {
+  [key: string]: TValue;
+}
+
+interface Context {
+  MessagingAPI: MessagingAPI;
+  $authUser?: User;
+  $channel?: Channel;
+  $message?: Message;
+}
 
 /**
  * MessagingHelper provides layer over Messaging API and utilities that simplify automation script writing
  *
  */
 export class Messaging {
-  constructor (ctx = {}) {
+  private MessagingAPI: MessagingAPI;
+  private $authUser?: User;
+  private $channel?: Channel;
+  private $message?: Message;
+
+  constructor (ctx: Context) {
```

Comment:
You could replace `Context` with `Partial<Messaging>`
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialt

Reply:
I've tried to do this, but i can't get it working. Looked at other code using Partial and tried it that way. No luck.

Reply:
What did you try and what does not work?

Reply:
Partial sets all the props to optional, so that isn't too good since we want the API to be set.